app/src/utils/apiOperations/notebooks.tsx

```diff
@@ -29,3 +36,102 @@ export const createNotebookFromTemplate = async (input: {
     {template_id: input.templateId, name: input.name} as PostCreateNotebookInput
   );
 };
+
+export interface RecordStatus extends PostRecordStatusResponse {
+  recordHashes: Record<string, string>;
+}
+
+/**
+ * Validate the sync status of records in a project
+ *
+ * @param projectId - the project id to validate
+ */
+export const validateSyncStatus = async ({
+  projectId,
+  username,
+  listingId,
+  currentStatus,
+}: {
+  projectId: ProjectID;
+  username: string;
+  listingId: string;
+  currentStatus: RecordStatus | undefined;
+}): Promise<RecordStatus> => {
+  // get the list of record ids from the project
+  const dataDb = await getDataDB(projectId);
+  const isOnline = window.navigator.onLine;
+  const emptyStatus = {
+    status: {},
+    recordHashes: {},
+  };
+
+  // get a list of record ids from the project
+  const records = await queryCouch({
+    db: dataDb,
+    index: RECORDS_INDEX,
+  });
+  const recordIds = records.map(r => r._id);
+  const audit = await getRecordListAudit({recordIds, dataDb});
+  let filteredAudit: Record<string, string> = {};
+
+  // now filter any records that we know are good from the last
+  // audit
+  if (currentStatus) {
+    for (const recordId of recordIds) {
+      // check the record if the hash has changed or the status was false last time
+      if (
+        audit[recordId] !== currentStatus.recordHashes[recordId] ||
+        !currentStatus.status[recordId]
+      ) {
+        filteredAudit[recordId] = audit[recordId];
+      }
+    }
+  } else {
+    filteredAudit = audit;
+  }
+
+  // if we're online, do the request
+  if (isOnline) {
+    if (Object.getOwnPropertyNames(filteredAudit).length > 0) {
+      const response = await FetchManager.post<PostRecordStatusResponse>(
+        listingId,
+        username,
+        `/api/notebooks/${projectId}/sync-status/`,
+        {record_map: filteredAudit} as PostRecordStatusInput
+      );
```

Comment:
can you put this into queries.ts or whatever it's called? Don't expose the API details here - that can be it's own method